File: Android Developing/Homework1/基本内容源代码/MyApplication3/app/src/main/java/com/example/myapplication/MainActivity.java
package com.example.myapplication;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // 显示信息
        findViewById(R.id.button2).setOnClickListener((e) -> {
            String str = ((EditText) findViewById(R.id.editTextTextPersonName)).getText().toString() + "今年"
                    + ((EditText) findViewById(R.id.editTextTextPersonName4)).getText().toString() + "岁了，身高是"
                    + ((EditText) findViewById(R.id.editTextTextPersonName3)).getText().toString() + "米。";
            ((TextView) findViewById(R.id.textView4)).setText(str);
        });

        // 删除信息
        findViewById(R.id.button3).setOnClickListener((e) -> {
            ((EditText) findViewById(R.id.editTextTextPersonName)).setText("");
            ((EditText) findViewById(R.id.editTextTextPersonName4)).setText("");
            ((EditText) findViewById(R.id.editTextTextPersonName3)).setText("");
            ((TextView) findViewById(R.id.textView4)).setText("");
        });
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        Toast.makeText(getApplicationContext(), "你点击了" + item.getTitle() + "菜单项", Toast.LENGTH_SHORT).show();
        return false;
    }
}

File: Android Developing/Homework2/程序二/MyAndroidHomework2ProviderTest/app/src/main/java/com/example/myandroidhomework2providertest/MainActivity.kt
package com.example.myandroidhomework2providertest

import android.content.DialogInterface
import android.content.Intent
import android.net.Uri
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.EditText
import androidx.appcompat.app.AlertDialog
import androidx.core.content.contentValuesOf
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    var ids = ArrayList<String?>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        addData.setOnClickListener {
            // 添加数据
            val uri = Uri.parse("content://xianfei.myandroidhomework2.provider/staff")
            val values = arrayOf(
                contentValuesOf(
                    "name" to "Tom",
                    "gender" to "male",
                    "department" to "computer",
                    "salary" to 5400
                ),contentValuesOf(
                    "name" to "Einstein",
                    "gender" to "male",
                    "department" to "computer",
                    "salary" to 4800
                ),contentValuesOf(
                    "name" to "Lily",
                    "gender" to "female",
                    "department" to "market",
                    "salary" to 5000
                ),contentValuesOf(
                    "name" to "Warner",
                    "gender" to "male",
                    "department" to "market"
                ),contentValuesOf(
                    "name" to "Napoleon",
                    "gender" to "male"
                )
            )
            values.forEach {
                val newUri = contentResolver.insert(uri, it)
                ids.add(newUri?.pathSegments?.get(1))
            }

        }

        queryData.setOnClickListener {
            // 查询数据
            textView.text = "id \t| name \t| gender \t| department \t| salary"
            val uri = Uri.parse("content://xianfei.myandroidhomework2.provider/staff")
            contentResolver.query(uri, null, null, null, null)?.apply {
                while (moveToNext()) {
                    val id = getInt(getColumnIndex("id"))
                    val name = getString(getColumnIndex("name"))
                    val gender = getString(getColumnIndex("gender"))
                    val department = getString(getColumnIndex("department"))
                    val salary = getInt(getColumnIndex("salary"))
                    textView.text =
                        textView.text.toString() + "\n $id \t| $name \t| $gender \t| $department \t| $salary";
                }
                close()
            }
        }

        updateData.setOnClickListener {
            // 更新数据
            ids.forEach {
                it?.let {
                    val uri = Uri.parse("content://xianfei.myandroidhomework2.provider/staff/$it")
//                val values = contentValuesOf("name" to "Tom",
//                "gender" to "male", "department" to "computer", "salary" to 1500)
                    val values = contentValuesOf("salary" to -100)
                    contentResolver.update(uri, values, null, null)
                }
            }

        }

        deleteData.setOnClickListener {
            // 删除数据
            ids.forEach {
                it?.let {
                    val uri = Uri.parse("content://xianfei.myandroidhomework2.provider/staff/$it")
                    contentResolver.delete(uri, null, null)
                }
            }
        }

        val intent = Intent(this, MyService::class.java)
        startService(intent) // 启动Service



    }


}
